rust/src/tests/rmdir.rs

```diff
@@ -29,5 +35,67 @@ fn changed_time_parent_success(ctx: &mut TestContext) {
         });
     });
 }
+
 // rmdir/01.t
 enotdir_comp_test_case!(rmdir);
+
+/// Dummy mountpoint to check that rmdir returns EBUSY when using it on a mountpoint.
+struct DummyMnt {
+    path: PathBuf,
+}
+
+impl DummyMnt {
+    fn new(ctx: &mut TestContext) -> anyhow::Result<Self> {
+        // We don't really care about a specific type of file system here, the directory just have to be a mount point
+        let from = ctx.create(crate::runner::context::FileType::Dir)?;
+        let path = ctx.create(crate::runner::context::FileType::Dir)?;
+        let mut mount = Command::new("mount");
+
+        if cfg!(target_os = "linux") {
+            mount.arg("--bind");
+        } else {
+            mount.args(["-t", "nullfs"]);
+        }
+
+        let result = mount.arg(&from).arg(&path).output()?;
+        if !std::thread::panicking() {
+            assert!(result.status.success());
+        }
+
+        Ok(Self { path })
+    }
+}
+
+impl Drop for DummyMnt {
+    fn drop(&mut self) {
+        let umount = Command::new("umount").arg(&self.path).output();
+        if !std::thread::panicking() {
+            assert!(matches!(umount, Ok(res) if res.status.success()));
+        }
+    }
+}
+
+#[cfg(target_os = "linux")]
+fn has_mount_cap(_: &Config, _: &Path) -> anyhow::Result<()> {
+    use caps::{has_cap, CapSet, Capability};
+
+    if !has_cap(None, CapSet::Effective, Capability::CAP_SYS_ADMIN)? {
+        anyhow::bail!("process doesn't have the CAP_SYS_ADMIN cap to mount the dummy file system")
+    }
+
+    Ok(())
+}
+
+#[cfg(not(target_os = "linux"))]
+fn has_mount_cap(_: &Config, _: &Path) -> anyhow::Result<()> {
+    Ok(())
+}
+
+crate::test_case! {
+    /// rmdir return EBUSY if the directory to be removed is the mount point for a mounted file system
+    ebusy, root; has_mount_cap
```

Comment:
On Linux, a non-root process can posses the `CAP_SYS_ADMIN` capability.  So the guard is sufficient; you don't need to check for root.  And on FreeBSD, you can mount a file system if the `vfs.usermount` sysctl is set.  So I suggest removing the root check here, and adding a check similar to Nix's for the FreeBSD implementation of `has_mount_cap`.
https://github.com/nix-rust/nix/blob/76f04df295c47ad5c9b74f67a37854b2a66137c9/test/common/mod.rs#L38